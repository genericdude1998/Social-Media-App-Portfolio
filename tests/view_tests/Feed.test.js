import React from 'react'
import { shallow } from 'enzyme';
import ConnectedFeed from '../../src/view/Feed/Feed';
import {mockPosts} from '../mockValues';

jest.spyOn(React, 'useEffect').mockImplementation(f => f());
const mockOnFetchPosts = jest.fn();

let wrapper = shallow(<ConnectedFeed.WrappedComponent
    posts={mockPosts}
    onFetchPosts={mockOnFetchPosts}
/>);

describe('Feed', () => {
    it('should call OnFetchPosts with expected params', () => {
        expect(mockOnFetchPosts).toHaveBeenCalled();
    });
    it('should match snapshot if list is not undefined', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should match snapshot if list is undefined', () => {
        let wrapper = shallow(<ConnectedFeed.WrappedComponent
            posts={undefined}
            onFetchPosts={mockOnFetchPosts}
        />);
        expect(wrapper).toMatchSnapshot();
    });
});