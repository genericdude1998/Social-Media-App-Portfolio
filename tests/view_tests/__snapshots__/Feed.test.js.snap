// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed should match snapshot if list is not undefined 1`] = `
<div
  className="baseFeed"
>
  <ul>
    <li
      key="1"
    >
      <Post
        content="mock_content"
        date="mock_date"
        id="1"
        name="mock_name"
      />
    </li>
    <li
      key="2"
    >
      <Post
        content="mock_content"
        date="mock_date"
        id="2"
        name="mock_name"
      />
    </li>
    <li
      key="3"
    >
      <Post
        content="mock_content"
        date="mock_date"
        id="3"
        name="mock_name"
      />
    </li>
  </ul>
</div>
`;

exports[`Feed should match snapshot if list is undefined 1`] = `
<div
  className="baseFeed"
>
  <ul>
    <h1>
      Loading
    </h1>
  </ul>
</div>
`;
